lib/broadway_cloud_pub_sub/acknowledger.ex

```diff
@@ -102,17 +52,15 @@ defmodule BroadwayCloudPubSub.Acknowledger do
 
   @impl Acknowledger
   def configure(_ack_ref, ack_data, options) do
-    options = assert_valid_config!(options)
-    ack_data = Map.merge(ack_data, Map.new(options))
-    {:ok, ack_data}
-  end
-
-  defp assert_valid_config!(options) do
-    Enum.map(options, fn
-      {:on_success, value} -> {:on_success, validate_option!(:on_success, value)}
-      {:on_failure, value} -> {:on_failure, validate_option!(:on_failure, value)}
-      {other, _value} -> raise ArgumentError, "unsupported configure option #{inspect(other)}"
-    end)
+    case NimbleOptions.validate(options, Options.acknowledger_definition()) do
+      {:ok, opts} ->
+        ack_data = Map.merge(ack_data, Map.new(opts))
+        {:ok, ack_data}
+
+      {:error, error} ->
+        raise ArgumentError,
+              Options.format_error(error, __MODULE__, :configure, 3)
```

Comment:
this is totally fine as is, just curious if there were limitations in the exception made by validate!/2 that made you write a custom one?

Reply:
😱 I never replied to this! Initially wrote the custom error formatter because I thought it would help clarify _where_ the config error occurred, since we validate on both the Producer and the Acknowledger, but on further thoughts it just added more complication.

I removed the custom formatting on the updated PR:

- #82